pycyphal/transport/udp/_socket_reader.py

```diff
@@ -204,19 +216,30 @@ def _dispatch_frame(
                 else:
                     handled = True
                     try:
-                        callback(timestamp, source_node_id, frame)
+                        callback(timestamp, frame)
                     except Exception as ex:  # pragma: no cover
                         _logger.exception("%r: Unhandled exception in the listener for node-ID %r: %s", self, key, ex)
 
-        # Update the statistics.
+        # Handle anonymous frames
+        elif anonymous_frame:
+            try:
+                callback = self._listeners[None]
+            except LookupError:
+                pass
+            else:
+                handled = True
+                try:
+                    callback(timestamp, frame)
+                except Exception as ex:
+                    _logger.exception("%r: Unhandled exception in the listenere for anonymous node-ID: %s", self, ex)
```

Comment:
This seems like code duplication to me, can we not handle both in the branch above?